src/NuGet.Core/NuGet.Build.Tasks.Pack.Library/Pack.targets

```diff
@@ -56,6 +56,16 @@ Copyright (c) .NET Foundation. All rights reserved.
           DependsOnTargets="Pack">
   </Target>
 
+  <Target Name="_CleanPackageFiles"
+          AfterTargets="Clean"
```

Comment:
I actually talked to @Rainersigwald this morning, and he said that this approach is fine too.